### ISSUE TITLE
Simple dropdown task: the select menu traps focus when open

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#2837]
## Package
lib-classifier

## Describe the bug
When a simple dropdown is open, it traps focus and Tab does not close the menu and move on to the next focusable item in the page.

## To Reproduce
Tab into the dropdown menu on this test workflow, select an item then tab to the Done button.
https://frontend.preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify/workflow/3489?env=staging&demo=true

Compare that behaviour with the expected behaviour of the HTML `<select>` element here:
https://codepen.io/eatyourgreens/full/zYPEWQq

## Expected behavior
Tab should select an option and move on to the next focusable element in the page.

## Additional context
This is the same Grommet bug as #2183, fixed in #2184 by adding `focusTrap: false` to select menus.

**Comment by shaunanoordin** [comment]
@eatyourgreens I've tested [your example on CodePen](https://codepen.io/eatyourgreens/full/zYPEWQq) and I'm noting different behaviours across browsers.

Testing keyboard interaction for a `<select>` element:
- When closed, I can tab-navigate to/from the element.
- When the element is closed and has focus, I can press the Up/Down key to open it.
- When open and has focus, I can press Up/Down to navigate between options ; Enter to select/confirm an option and close ; or Esc to close.
- When open and has focus, pressing the Tab key...
  - ♠️ Chrome 98 / Safari 15: ... **does nothing.** (Element still has focus and is open)
  - ♣️ Firefox 97: ...closes the element. Element still has focus. (Same as Esc key)

I'm not sure what's the "proper" behaviour's supposed to be here. If there's no official W3 documentation on this, then I'm OK for us to be opinionated on how a dropdown/`<select>`-like should behave when you tab-out.

(Currently, on `master` (54fb90d843a30e6541e6b2a22225303c2e2ecb34), pressing the Tab key on a Simple Dropdown on Chrome sends the focus to the browser's navigation bar and makes it hard to regain focus on the Simple Dropdown again, so that's mildly borked in any case.)

**Comment by eatyourgreens** [comment]
WebAIM's[ list of standard keyboard interactions](https://webaim.org/techniques/keyboard/#testing) doesn't even mention Tab for select menus. Seems like it is up to the individual browser manufacturers, or implementors of JS widgets, how they want to handle Tab navigation. As a Firefox user, I must admit I find it very handy to be able to use Tab to select an option and move on.

**Comment by eatyourgreens** [comment]
@shaunanoordin something else I'm also seeing, in the DOM inspector when the menu is open: each menu option is a `<button>` with `tabindex="-1"`. That means that those buttons can't be focussed with Tab at all.

**Comment by eatyourgreens** [comment]
Updating this to add the WAI-ARIA 1.2 combo box example, which does select an option and close on tab. 
https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html

**Comment by eatyourgreens** [comment]
I've also set up a couple of codepens with the menu examples from WAI-ARIA 1.2. We can use these as a reference for our implementations of custom widgets.

Combo-box with autocomplete:
https://codepen.io/eatyourgreens/pen/abEQrBg

Simple select menu:
https://codepen.io/eatyourgreens/pen/GRywaMm

**Comment by goplayoutside3** [comment]
Linking a discussion about why we stuck with Grommet in this task component: https://github.com/zooniverse/front-end-monorepo/pull/5382

Is this still an issue to keep open? With my keyboard, when I open the simple select menu [codepen](https://codepen.io/eatyourgreens/full/zYPEWQq) from the description, I have to select an option via Enter to close the menu before I can tab to the next focusable element. The [test project](https://frontend.preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify/workflow/3489?env=staging&demo=true) task has the same behavior - from the open menu I have to select an option via Enter and then I'm free to tab to the next focusable element.

**Comment by eatyourgreens** [comment]
>With my keyboard, when I open the simple select menu [codepen](https://codepen.io/eatyourgreens/full/zYPEWQq) from the description, I have to select an option via Enter to close the menu before I can tab to the next focusable element. The [test project](https://frontend.preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify/workflow/3489?env=staging&demo=true) task has the same behavior - from the open menu I have to select an option via Enter and then I'm free to tab to the next focusable element.

Having to select an option with Enter is the bug. Volunteers, particularly in Firefox, expect Tab to select an option and move to the next menu. That’s the behaviour that they’re accustomed to from NfN projects on PFE.

**Comment by eatyourgreens** [comment]
Another complication, when testing how volunteers use select menus, is that dropdown menus are handled differently by Windows and MacOS. Most volunteers are on Windows, so accustomed to select menus being comboboxes:

>One of the largest roadblocks in creating an accessible cross-platform select component is that different platforms have fundamental disagreements on what a select component actually is. If you inspect the underlying [Accessibility API mappings](https://alistapart.com/article/semantics-to-screen-readers/) on Windows and macOS for a <select> element, you’ll notice that it maps to a combobox on Windows and a popup button (which opens a menu) on macOS.

https://www.24a11y.com/2019/select-your-poison/

**Comment by eatyourgreens** [comment]
Here's the corresponding Grommet issue, with some more discussion of the problems that Zooniverse volunteers were running into:
- https://github.com/grommet/grommet/issues/5896

**Comment by goplayoutside3** [comment]
Thanks for the clarification 👍 will leave this open for now!
